pkg/actions/iam/update.go

```diff
@@ -0,0 +1,51 @@
+package iam
+
+import (
+	"fmt"
+
+	"github.com/kris-nova/logger"
+
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+
+	"github.com/weaveworks/eksctl/pkg/utils/tasks"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+)
+
+func (a *Manager) UpdateIAMServiceAccounts(iamServiceAccounts []*api.ClusterIAMServiceAccount, plan bool) error {
+	var nonExistingSAs []string
+	updateTasks := &tasks.TaskTree{Parallel: true}
+
+	for _, iamServiceAccount := range iamServiceAccounts {
+		stackName := makeIAMServiceAccountStackName(a.clusterName, iamServiceAccount.Namespace, iamServiceAccount.Name)
+		stacks, err := a.stackManager.ListStacksMatching(stackName)
+		if err != nil {
+			return err
+		}
+
+		if len(stacks) == 0 {
+			logger.Info("Cannot update IAMServiceAccount %s/%s as it does not exist", iamServiceAccount.Namespace, iamServiceAccount.Name)
+			nonExistingSAs = append(nonExistingSAs, fmt.Sprintf("%s/%s", iamServiceAccount.Namespace, iamServiceAccount.Name))
```

Comment:
shouldn't this be handled by the exclude/include filters?

Reply:
I didn't spot that we used filtering for `create` iamserviceaccount, I'll update it so we use that here too